src/rename.c

```diff
@@ -8,12 +8,28 @@
 #endif
 
 static const int rename_flags_table[] = {
-  RENAME_NOREPLACE, RENAME_EXCHANGE, RENAME_WHITEOUT,
+  RENAME_NOREPLACE, /* 0 */
+  RENAME_EXCHANGE, /* 1 */
+  RENAME_WHITEOUT, /* 2 */
 };
 
+#define RENAME_WHITEOUT_INDEX 2
+
+static void check_flag_list(value list)
+{
+  for (/*nothing*/; list != Val_emptylist; list = Field(list, 1))
+  {
+#if !defined(EXTUNIX_HAVE_RENAME_WHITEOUT)
+    if (RENAME_WHITEOUT_INDEX == Int_val(Field(list, 0)))
+      caml_raise_with_string(*caml_named_value("ExtUnix.Not_available"), "renameat2 RENAME_WHITEOUT");
```

Comment:
it is not very defined i think but lets have the argument to Not_available be exactly the symbol that is not available, so just "RENAME_WHITEOUT"